tsweb/tsweb.go

```diff
@@ -463,3 +497,33 @@ func writeMemstats(w io.Writer, ms *runtime.MemStats) {
 	c("frees", ms.Frees, "cumulative count of heap objects freed")
 	c("num_gc", uint64(ms.NumGC), "number of completed GC cycles")
 }
+
+func foreachExportedStructField(rv reflect.Value, f func(fieldOrJSONName, promTyp string, rv reflect.Value)) {
+	t := rv.Type()
+	for i, n := 0, t.NumField(); i < n; i++ {
+		sf := t.Field(i)
+		name := sf.Name
+		if v := sf.Tag.Get("json"); v != "" {
+			if i := strings.Index(v, ","); i != -1 {
+				v = v[:i]
+			}
+			if v != "" {
```

Comment:
`v != "" && v != "-"`

Reply:
Stepping back a little. Do we really want to use the name from the `json` tag?

```go
type MyStruct {
    FooBar int    `json:"bar"`
    FizzBuzz int  `json:"-"`
}
```
For the above struct, `FooBar` will be named `bar`, but `FizzBuzz` will be named `FizzBuzz`. Seems a bit inconsistent.

Reply:
Done, with new test.

Reply:
Yes, because we're replacing a big JSON-encoding struct of struct of struct now and they all have struct tag names today.

Reply:
SGTM.